client/flutter/lib/api/user_preferences.dart

```diff
@@ -68,7 +70,9 @@ class UserPreferences {
 
     // Create if not found
     if (uuid == null) {
-      uuid = Uuid().v4();
+      uuid = Uuid(options: {
+        'grng': UuidUtil.cryptoRNG
+      }).v4();
```

Comment:
The other option we were going to suggest was having a server endpoint that generates the UUIDs and ensures they are unique and then returns them to clients, so that the server controls the entropy source and ensures they are generated strong and uniquely. Happy to discuss further in ticket in case that is of interest.

Reply:
I think we should definitely discuss this further for future versions!

Reply:
The point of UUIDs is they can be generated independently, without an authority like a single server.  As long as UUIDs are the sole form of auth - why does it matter who generates them?

Reply:
Advay is right. Phones have plenty of entropy—their security already depends upon it. Not making another round trip to the server means users can start the app without a network connection.

Reply:
Indeed - makes sense to avoid the round trip to let the app start without a network connection.